### PR TITLE
Added friends list functionality.

### PR DESCRIPTION
**Pull request by sbaum23**
Friends load from database on login and add/removing friends also saves to the db.  When a friend logs on, the player should see the status change to online but it's hard to test that completely right now.  Changing the status to offline when a friend logs off is not yet implemented.

I had to implement a couple methods in WorldManager to be able to query for connected clients.

### CODE REVIEW

File: Source/ACE/Network/GameEvent/Events/GameEventFriendsListUpdate.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+using ACE.Managers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ACE.Network.GameEvent
+{
+    class GameEventFriendsListUpdate : GameEventPacket
```
Comment (Mogwai-AC) [inline Source/ACE/Network/GameEvent/Events/GameEventFriendsListUpdate.cs]: missing scope declaration (public/private/internal)

Comment (sbaum23) [inline Source/ACE/Network/GameEvent/Events/GameEventFriendsListUpdate.cs]: Thanks for the review.  I've fixed and committed.

File: Source/ACE/Network/GameAction/GameActionOpcode.cs [hunk 2]
```diff
@@ -3,13 +3,16 @@
     // order by opcode name
```
Comment (Rawaho) [inline Source/ACE/Network/GameAction/GameActionOpcode.cs]: Clearly says order by name ;)

Comment (sbaum23) [inline Source/ACE/Network/GameAction/GameActionOpcode.cs]: I clearly missed that!  I've made another commit and fixed the ordering of opcodes in both GameEventOpcode and GameActionOpCode.

**Comment by sbaum23** [comment]
Closing this PR because so much stuff had changed since this was submitted.  Just opened a new PR.

**Pull request closed without merging**